core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncSigningStage.java

```diff
@@ -128,8 +129,17 @@ private <T extends Identity> CompletableFuture<SdkHttpFullRequest> doSraSign(Sdk
     }
 
     private static void updateAsyncRequestBodyInContexts(RequestExecutionContext context, AsyncSignedRequest signedRequest) {
-        AsyncRequestBody newAsyncRequestBody = signedRequest.payload().isPresent() ?
-                                               AsyncRequestBody.fromPublisher(signedRequest.payload().get()) : null;
+        AsyncRequestBody newAsyncRequestBody;
+        if (signedRequest.payload().isPresent()) {
+            Publisher<ByteBuffer> signedPayload = signedRequest.payload().get();
+            if (signedPayload instanceof AsyncRequestBody) {
+                newAsyncRequestBody = (AsyncRequestBody) signedPayload;
+            } else {
+                newAsyncRequestBody = AsyncRequestBody.fromPublisher(signedPayload);
+            }
+        } else {
+            newAsyncRequestBody = null;
+        }
```

Comment:
nit: can we just assign `null` in line 132 and remove this branch?

Reply:
yeah I had started there, but this variable needs to be final or effectively final in the lambda expression later in
```
        executionContext.interceptorContext(executionContext.interceptorContext()
                                                            .copy(b -> b.asyncRequestBody(newAsyncRequestBody)));
```